### PR TITLE
source-google-forms contribution from bala-ceg

### LINKED ISSUE #47331: [Connector-Request] Google Forms
### Topic

_No response_

### Relevant information

This is a suggestion to create a connector for Google Forms using the UI Connector Builder.
The website for the connector is https://developers.google.com/forms/api/reference/rest/?apix=true

If you want to try, claim this issue and start working on it following the steps below.

**Steps:**
Comment in the issue and wait to be assigned to start working on it.
Map the API endpoints (get approval to move to step 3).
Describe steps to get credentials.
Create the connector using the UI Builder.
Airbyte doesn't have sandbox credentials for this connector. You must have access/credentials to the service provider to create the connector.

This is a suggestion, and there may be cases where creating the connector using the Builder won't be possible. To minimize the risk of investing a lot of time directly in creating the connector, we strongly recommend following the steps above.

### PR DESCRIPTION
**Pull request by bala-ceg**
## What

This PR adds a new connector for source Google Forms (source-google-forms).

Contributor-provided connector description:

Google Forms is a free online tool from Google that allows users to create custom surveys, quizzes, and forms. It enables easy collection and organization of data by automating responses into a connected Google Sheets spreadsheet. With Google Forms, you can design forms with various question types, share them via email or links, and track responses in real-time, making it ideal for feedback collection, event registration, or educational assessments.
## Details

<details><summary>Expand for list of spec properties and streams</summary>

### Configuration

| Input | Type | Description | Default Value |
|-------|------|-------------|---------------|
| `client_id` | `string` | Client ID.  |  |
| `client_secret` | `string` | Client secret.  |  |
| `client_refresh_token` | `string` | Refresh token.  |  |
| `form_id` | `array` | form_id.  |  |

### Streams
| Stream Name | Primary Key | Pagination | Supports Full Sync | Supports Incremental |
|-------------|-------------|------------|---------------------|----------------------|
| forms |  | No pagination | ✅ |  ❌  |
| form_responses | responseId | DefaultPaginator | ✅ |  ❌  |

</details>

## Reviewer checklist
- [ ] Run `/format-fix` to resolve any formatting errors
- [ ] Help the contributor upload an icon that follows our icon design guidelines
- [ ] Click into the CI workflows that wait for a maintainer to run them, which should trigger CI runs

### CODE REVIEW

File: airbyte-integrations/connectors/source-google-forms/manifest.yaml [hunk 1]
```diff
@@ -0,0 +1,401 @@
...
+definitions:
+  streams:
+    forms:
+      type: DeclarativeStream
+      name: forms
+      retriever:
+        type: SimpleRetriever
+        requester:
+          $ref: "#/definitions/base_requester"
+          path: "{{config[\"formId\"]}}"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-google-forms/manifest.yaml:28]: Make this a paremeter request:
image

File: airbyte-integrations/connectors/source-google-forms/manifest.yaml [hunk 2]
```diff
@@ -0,0 +1,401 @@
...
+          type: RecordSelector
+          extractor:
+            type: DpathExtractor
+            field_path: []
+      schema_loader:
+        type: InlineSchemaLoader
+        schema:
+          $ref: "#/schemas/forms"
+    form_responses:
+      type: DeclarativeStream
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-google-forms/manifest.yaml:40]: Include transformation to know from what form the response is from.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
This is a very simple connector with 2 endpoints only but can be helpful to capture survey responses. Make sure it works with multiple forms.

**Comment by marcosmarxm** [comment]
Moving next review to Nov 11th

**Comment by bala-ceg** [comment]
@marcosmarxm, I addressed the review comments, added the icon.svg image and google forms api doesn't mention anything about pagination. Can you please review this connector ? Thanks

**Review by marcosmarxm — APPROVED** [review]

**Pull request merged**